pkg/controllers/controlplanemachineset/cluster_operator_test.go

```diff
@@ -128,29 +148,211 @@ var _ = Describe("Cluster Operator Status with a running controller", func() {
 				})).Should(Succeed())
 			})
 
-			PIt("Set the cluster operator available", func() {
-				co := resourcebuilder.ClusterOperator().WithName(operatorName).Build()
-
-				Eventually(komega.Object(co)).Should(HaveField(".Status.Conditions", test.MatchClusterOperatorStatusConditions([]configv1.ClusterOperatorStatusCondition{
+			It("Set the cluster operator available", func() {
+				Eventually(komega.Object(co)).Should(HaveField("Status.Conditions", test.MatchClusterOperatorStatusConditions([]configv1.ClusterOperatorStatusCondition{
 					{
-						Type:   configv1.OperatorAvailable,
-						Status: configv1.ConditionTrue,
+						Type:    configv1.OperatorAvailable,
+						Status:  configv1.ConditionTrue,
+						Reason:  reasonAsExpected,
+						Message: "cluster operator is available",
 					},
 					{
 						Type:   configv1.OperatorProgressing,
 						Status: configv1.ConditionFalse,
+						Reason: reasonAsExpected,
 					},
 					{
 						Type:   configv1.OperatorDegraded,
 						Status: configv1.ConditionFalse,
+						Reason: reasonAsExpected,
 					},
 					{
-						Type:   configv1.OperatorUpgradeable,
-						Status: configv1.ConditionTrue,
+						Type:    configv1.OperatorUpgradeable,
+						Status:  configv1.ConditionTrue,
+						Reason:  reasonAsExpected,
+						Message: "cluster operator is upgradable",
 					},
 				})))
 			})
 		})
+	})
+})
+
+var _ = Describe("Cluster Operator Status", func() {
+	const operatorName = "control-plane-machine-set"
+	var co *configv1.ClusterOperator
+	var reconciler *ControlPlaneMachineSetReconciler
+	var logger test.TestLogger
+	var namespaceName string
+
+	cpmsBuilder := resourcebuilder.ControlPlaneMachineSet().WithName(clusterControlPlaneMachineSetName).WithNamespace(namespaceName)
```

Comment:
You need to construct this in the before each else the namespaceName is just the empty string
```suggestion
	var cpmsBuilder resourcebuilder.ControlPlaneMachineSetBuilder 
```
Then in the before each 
```
cpmsBuilder = resourcebuilder.ControlPlaneMachineSet().WithName(clusterControlPlaneMachineSetName).WithNamespace(namespaceName)
```